Source/GSAutoLayoutEngine.m

```diff
@@ -199,8 +346,280 @@ - (void) removeConstraints: (NSArray *)constraints
 - (GSCSConstraint *) solverConstraintForConstraint:
     (NSLayoutConstraint *)constraint
 {
-  // FIXME Map NSLayoutConstraint to GSCSConstraint
-  return AUTORELEASE([[GSCSConstraint alloc] init]);
+  if ([constraint secondItem] == nil)
+    {
+      return [self solverConstraintForNonRelationalConstraint: constraint];
+    }
+  else
+    {
+      return [self solverConstraintForRelationalConstraint: constraint];
+    }
+}
+
+- (GSCSConstraint *) solverConstraintForNonRelationalConstraint:
+    (NSLayoutConstraint *)constraint
+{
+  GSCSVariable *firstItemConstraintVariable =
+    [self variableForView: [constraint firstItem]
+             andAttribute: (GSLayoutAttribute)[constraint firstAttribute]];
+  GSCSConstraint *newConstraint;
+  switch ([constraint relation])
+    {
+    case NSLayoutRelationLessThanOrEqual:
+      newConstraint = [GSCSConstraint constraintWithLeftVariable: firstItemConstraintVariable operator: GSCSConstraintOperatorLessThanOrEqual rightConstant: [constraint constant]];
+      break;
+    case NSLayoutRelationEqual:
+      newConstraint =  [GSCSConstraint constraintWithLeftVariable: firstItemConstraintVariable operator: GSCSConstraintOperatorEqual rightConstant: [constraint constant]];
+      break;
+    case NSLayoutRelationGreaterThanOrEqual:
+      newConstraint = [GSCSConstraint constraintWithLeftVariable: firstItemConstraintVariable operator: GSCSConstraintOperationGreaterThanOrEqual rightConstant: [constraint constant]];
+      break;
+    }
+  GSCSStrength *strength =
+    [[GSCSStrength alloc] initWithName: nil strength: [constraint priority]];
+  [newConstraint setStrength: strength];
+  RELEASE(strength);
+
+  return newConstraint;
+}
+
+- (GSCSConstraint *) solverConstraintForRelationalConstraint:
+  (NSLayoutConstraint *)constraint
+{
+  GSCSVariable *firstItemConstraintVariable =
+    [self variableForView: [constraint firstItem]
+             andAttribute: (GSLayoutAttribute)[constraint firstAttribute]];
+  GSCSVariable *secondItemConstraintVariable =
+    [self variableForView: [constraint secondItem]
+             andAttribute: (GSLayoutAttribute)[constraint secondAttribute]];
+
+  CGFloat multiplier = [constraint multiplier];
+
+  GSCSConstraintOperator op = GSCSConstraintOperatorEqual;
+  switch ([constraint relation])
```

Comment:
You could have a map for these values to replace the switch in this and the previous method.